### PR TITLE
Switch package manager to pnpm

### PR DESCRIPTION
**Pull request by dgeb**
pnpm properly handles `peerDependencies`, which yarn v1 does not. Proper management of peer dependencies will be critical to support usage of `@glimmer/validator`, which can not be duplicated at run time. A subsequent PR will bring in this peer dep as a part of a refactor of `TrackedObject`.

### CODE REVIEW

**Comment by SergeAstapov** [comment]
@dgeb as I overall prefer pnpm over yarn and love ergonomics it brings to monorepos, I'm curios how switching to pnpm in this repo can help solving issues with peer deps for consuming apps?

**Comment by dgeb** [comment]
> I'm curios how switching to pnpm in this repo can help solving issues with peer deps for consuming apps?

@SergeAstapov Proper handling of peer deps is essential for packages such as `@glimmer/validator`, which can only be installed/registered once globally. In fact, there is a [global check](https://github.com/glimmerjs/glimmer-vm/blob/master/packages/@glimmer/validator/index.ts#L9) in place to specifically ensure this. This check will fail when using yarn and adding a peer dep of `@glimmer/validator` to the `tracked-built-ins` addon package. This happens because yarn incorrectly hoists this dep to the root `node_modules`, and it is resolved multiple times at run-time when run together with the `test-app`.

On the other hand, pnpm _does not_ hoist this peer dep to the root `node_modules`, so it will not be resolved and registered multiple times when run together with `test-app` (or any other client app).

**Comment by SergeAstapov** [comment]
@dgeb that makes sense. so idea is to get the test setup in this addon to properly reflect the real-world consuming app use case

**Comment by SergeAstapov** [comment]
@dgeb you may submit https://github.com/tracked-tools/tracked-built-ins/pull/393/files#diff-4138d264b0db08dc8194a4ed0f79c64e2ad0c62e0787d0b1e7fd73a22a56755a in separate PR, we'll merge and voila - no approvals would be needed

**Comment by dgeb** [comment]
@SergeAstapov Great idea, thanks! See #394

**Comment by dgeb** [comment]
@SergeAstapov I believe this is ready when you are. Thanks in advance (and no rush)!

File: addon/babel.config.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "presets": [["@babel/preset-typescript"]],
   "plugins": [
-    "@embroider/addon-dev/template-colocation-plugin",
+    "@babel/plugin-proposal-class-properties",
```
Comment (SergeAstapov) [inline addon/babel.config.json:4]: @dgeb do we need to compile away class properties? I think this could be delegated to the host app but I may be wrong

**Review by SergeAstapov — APPROVED** [review]

File: addon/babel.config.json (hunk 1 above)
Comment (SergeAstapov) [reply addon/babel.config.json:4]: this worth separate discussion indeed. landing for now

**Comment by SergeAstapov** [comment]
Thank you @dgeb! I love CI speed improvements this gives ⚡

**Comment by SergeAstapov** [comment]
note for myself: replace dependabot with renovate for pnpm support

File: addon/babel.config.json (hunk 1 above)
Comment (dgeb) [reply addon/babel.config.json:4]: Good catch @SergeAstapov - this slipped in when I copied from another config. I'll PR its removal.

**Comment by dgeb** [comment]
> next step: replace dependabot with renovate for pnpm support

@SergeAstapov In a brief look at similar pnpm-based addons that use renovate, I came across https://github.com/NullVoxPopuli/ember-popperjs/blob/main/.github/renovate.json5 from @NullVoxPopuli.

Does that look like a good starting point for this project or do you have a renovate template / settings you would prefer?

**Comment by SergeAstapov** [comment]
@dgeb https://github.com/NullVoxPopuli/ember-popperjs/blob/main/.github/renovate.json5 seem good, just IMO might over-complicated. I wonder if we can start with something simpler, e.g. like https://github.com/mainmatter/renovate-config/blob/main/default.json5 and adjust as we go considering how much automation we want/need here?

**Comment by SergeAstapov** [comment]
just another idea - maybe it worth to add renovate-config to addon blueprint when it has github as CI provider? that way, it could be iterated over by whole community to provide good overall defaults as I think majority of addons have either renovate or dependabot integrated
indeed would need an RFC for that

**Comment by dgeb** [comment]
@SergeAstapov good call on both counts 👍

I'll prep a PR with the simpler config from mainmatter to start.

**Comment by NullVoxPopuli** [comment]
> over-complicated.

there is a reason for some of that complication ;) 

renovate, by default, does breaking-change merges with the dependencies.
only devDependencies of a library are safe to change willy-nilly

**Comment by dgeb** [comment]
> renovate, by default, does breaking-change merges with the dependencies.

@NullVoxPopuli ah, thanks for the heads up! Looks like the `rangeStrategy` option is particularly important here.

I agree with @SergeAstapov that it would be ideal if we (i.e. the ember community) should iterate on a common config to use in the addon blueprint. Let's discuss this more soon (probably in an open issue and/or on discord).

**Pull request merged**